pkg/minishift/registration/redhat.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"strings"
+	"bufio"
+	"fmt"
+	"os"
+	"syscall"
+	"golang.org/x/crypto/ssh/terminal"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+
+func init() {
+	Register("Redhat", &RegisteredRegistrator{
+		New: NewRedhatRegistrator,
+	})
+}
+
+func NewRedhatRegistrator(d drivers.Driver) Registrator {
+	return &RedhatRegistrator{
+		SSHCommander:   GenericSSHCommander{Driver: d},
+		Driver: d,
+	}
+}
+
+type RedhatRegistrator struct {
+	SSHCommander
+	Driver drivers.Driver
+}
+
+type GenericSSHCommander struct {
+	Driver drivers.Driver
+}
+
+func (sshCmder GenericSSHCommander) SSHCommand(args string) (string, error) {
+	return drivers.RunSSHCommandFromDriver(sshCmder.Driver, args)
+}
+
+func credentials() (string, string) {
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Print("Enter Username: ")
+	username, _ := reader.ReadString('\n')
+
+	fmt.Print("Enter Password: ")
+	bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err == nil {
+		fmt.Println("\nPassword typed: " + string(bytePassword))
+	}
+	password := string(bytePassword)
+
+	return strings.TrimSpace(username), strings.TrimSpace(password)
+}
+
+func (registrator *RedhatRegistrator) CompatibleWithDistribution() bool {
+	if _, err := registrator.SSHCommand("sudo subscription-manager version"); err != nil {
+		return false
+	} else {
+		return true
+	}
+}
+
+func (registrator *RedhatRegistrator) GetRegister() error {
+	if output, err := registrator.SSHCommand("sudo subscription-manager version"); err != nil {
+		return err
+	} else
+	{
+		if strings.Contains(output, "not registered") {
+			username, password := credentials()
+			if _, err := registrator.SSHCommand(fmt.Sprintf(
+				"sudo subscription-manager register --username %s --password %s --auto-attach",
+			username, password)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (registrator *RedhatRegistrator) GetUnregister() error {
+	if output, err := registrator.SSHCommand("sudo subscription-manager version"); err != nil {
+		return err
+	} else
+	{
+		if ! strings.Contains(output, "not registered") {
+			if _, err := registrator.SSHCommand(
+				"sudo subscription-manager unregister"); err != nil {
```

Comment:
Same as above, we should use proxy options, so that user can register and un-register with proxy server.